### PR TITLE
Fix getSSOData not sending custom nonce/state

### PR DESCRIPTION
**Pull request by luisrudge**
fix https://github.com/auth0/lock/issues/1331

### CODE REVIEW

File: src/__tests__/core/web_api/p2_api.test.js [hunk 1]
```diff
@@ -79,6 +83,14 @@ describe('Auth0APIClient', () => {
...
         expect(client.authOpt.state).toBe('foo');
       });
+      it('should set nonce from options.nonce', () => {
+        const client = getClient({ nonce: 'foo' });
+        expect(client.authOpt.nonce).toBe('foo');
+      });
+      it('should set nonce from options.params.nonce', () => {
+        const client = getClient({ params: { nonce: 'foo' } });
+        expect(client.authOpt.nonce).toBe('foo');
+      });
```
Comment (lbalmaceda) [inline src/__tests__/core/web_api/p2_api.test.js]: what if both `nonce` are set, which one should win? Add that test and document this behavior somewhere.

File: src/__tests__/core/web_api/p2_api.test.js [hunk 2]
```diff
@@ -79,6 +83,14 @@ describe('Auth0APIClient', () => {
         const client = getClient({ params: { state: 'foo' } });
         expect(client.authOpt.state).toBe('foo');
       });
```
Comment (lbalmaceda) [inline src/__tests__/core/web_api/p2_api.test.js]: what if both `state` are set, which one should win? Add that test and document this behavior somewhere.

Comment (luisrudge) [reply src/__tests__/core/web_api/p2_api.test.js]: added

File: src/__tests__/core/web_api/p2_api.test.js (hunk 1 above)
Comment (luisrudge) [reply src/__tests__/core/web_api/p2_api.test.js]: added

**Pull request merged**